invenio_app_rdm/utils/files.py

```
@@ -0,0 +1,125 @@
+        if deposit and deposit.json:
+            entry["deposit"] = deposit.json
+        else:
+            record = get_record_from_bucket(o.bucket_id)
+            if record and record.json:
+                entry["record"] = record.json
+    return entry
+
+
+def get_record_from_bucket(bucket_id):
+    """Retrieve a record from a bucket id."""
+    return RDMRecordMetadata.query.filter_by(bucket_id=bucket_id).one_or_none()
+
+
+def get_deposit_from_bucket(bucket_id):
+    """Retrieve a deposit from a bucket id."""
+    return RDMDraftMetadata.query.filter_by(bucket_id=bucket_id).one_or_none()
+
+
+def get_report_subject():
+    """Retrieves the integrity report e-mail's subject."""
+    subject = current_app.config.get("FILES_INTEGRITY_REPORT_SUBJECT")
+    if not subject:
+        subject = "Files integrity report"
+    subject += " [{}]".format(datetime.now())
+    return subject
+
+
+def get_report_template():
+    """Retrieves the template name to be used in the files integrity report."""
+    tpl = current_app.config.get("FILES_INTEGRITY_REPORT_TEMPLATE")
+    # Use a default template
+    if not tpl:
+        tpl = "invenio_app_rdm/emails/files_integrity_report.html"
+    return tpl
+
+
+def render_report_email(files):
+    """Renders the report e-mail from a list of files."""
+    entries = generate_integrity_report_entries(files)
+    tpl = get_report_template()
+    context = {"entries": entries}
+    return render_email_from_context(tpl, context)
+
+
+def send_integrity_report_email(files):
+    """Sends an e-mail with a report on the provided files.
+
+    APP configs ('SECURITY_EMAIL_SENDER', 'ADMIN_EMAIL_RECIPIENT') must be set.
+    Oherwise a warning is logged.
+    """
+    try:
+        if files:
+            subject = get_report_subject()
+            body = render_report_email(files)
+            sender = current_app.config.get("SECURITY_EMAIL_SENDER")
+            admin_email = current_app.config.get("ADMIN_EMAIL_RECIPIENT")
```

There are too many `dict.get()` around: IMO, for what is required, we should use `dict[]` and fail if missing.
Better fail-fast whenever possible. Can you check all the occurrences?